### PR TITLE
BlackRock changes

### PR DESCRIPTION
**Pull request by ghost**
When calling getAllUsers() the method queries for all Symphony user ids, and then creates a bunch of threads that each queries Symphony POD for the information about a user id. The code currently allows 5 seconds for all users to be returned. After 5 seconds it returns the subset of users that have been added; in BlackRock this means calls to getAllUsers() only returns one third of all users. Additionally the flag indicating if a given SymUser is active, e.g. their account is enabled, is not set.

This pull request:
* Leaves the default time for waiting for all user data to be returned (unchanged) at 5 seconds
* Allows the default time to be modified by setting SYMPHONY_GET_ALL_USERS_TIMEOUT as a VM argument, e.g. for 2 minutes -DSYMPHONY_GET_ALL_USERS_TIMEOUT=120
* Sets the active flag on SymUser when calling getAllUsersWithDetails()

### CODE REVIEW

File: symphony-client/src/main/java/org/symphonyoss/symphony/clients/impl/UsersClientImpl.java [hunk 1]
```diff
@@ -93,7 +91,7 @@ public UsersClientImpl(SymAuth symAuth, SymphonyClientConfig config, Client http
 
         apiClient.setBasePath(config.get(SymphonyClientConfigID.POD_URL));
 
-
+        getAllUsersTimeout = Long.valueOf(System.getProperty("SYMPHONY_GET_ALL_USERS_TIMEOUT", "5")).longValue();
```
Comment (pmonks) [inline symphony-client/src/main/java/org/symphonyoss/symphony/clients/impl/UsersClientImpl.java]: Using hardcoded JVM system properties for this kind of thing is rather brittle, and complicates the use of techniques such as dependency injection (i.e. Spring) for bootstrapping code.

A better alternative would be to add this new timeout configuration option to the [`SymphonyClientConfigID` enum](https://github.com/symphonyoss/symphony-java-client/blob/master/symphony-client/src/main/java/org/symphonyoss/client/SymphonyClientConfigID.java#L19), and then fix any fallout elsewhere (e.g. ensuring it also gets added to [`SymphonyClientConfigBuilder`](https://github.com/symphonyoss/symphony-java-client/blob/master/symphony-client/src/main/java/org/symphonyoss/client/SymphonyClientConfigBuilder.java).

Comment (ftbb) [reply symphony-client/src/main/java/org/symphonyoss/symphony/clients/impl/UsersClientImpl.java]: Somewhat old code.  Agree with @pmonks we need to add it into the SymphonyClientConfigID etc..   This makes it more consistent with SJC configuration management.

**Comment by blackrock-engineering** [comment]
Updated code in line with previous comments.

Thoughts?

**Comment by blackrock-engineering** [comment]
I will merge this pull request on Tuesday 16th October (2 weeks away). 

Please raise any concerns prior to this date.

**Pull request merged**